mat/cholesky.go

```diff
@@ -91,7 +91,7 @@ func (c *Cholesky) Factorize(a Symmetric) (ok bool) {
 // Reset resets the factorization so that it can be reused as the receiver of a
 // dimensionally restricted operation.
 func (c *Cholesky) Reset() {
-	if !c.isZero() {
+	if !c.chol.IsZero() {
```

Comment:
Does this if need to be there? Can't we just call Reset?

Reply:
Looking now, the original code is correct. `Reset` cannot be called on `nil`, otherwise it will panic when it attempt to update the fields (the code here will now too, when it checks). The `!isZero()` guards that. The transformation that I've applied elsewhere would make that clear.